packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3256 @@
+/// a slight delay, and hovering over other items will close that menu and open
+/// the newly hovered one. When those occur, [MenuBarMenu.onOpen], and
+/// [MenuBarMenu.onClose] are called, respectively.
+///
+/// {@tool dartpad}
+/// This example shows a [MenuBar] that contains a single top level menu,
+/// containing three items for "About", a checkbox menu item for showing a
+/// message, and "Quit". The items are identified with an enum value.
+///
+/// ** See code in examples/api/lib/material/menu_bar/menu_bar.0.dart **
+/// {@end-tool}
+///
+/// {@tool sample}
+/// This example shows a [MenuBar] that contains a simple menu for a desktop
+/// application. It is set up to be adaptive, so that on macOS it will use the
+/// system menu bar, and on other systems will use a Material [MenuBar].
+///
+/// On macOS, it will add the "About" and "Quit" system-provided menu items.
+///
+/// The menu items are all identified by an enum value (`MenuSelection`).
+///
+/// ** See code in examples/api/lib/material/menu_bar/menu_bar.1.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [MenuBarMenu], a menu item which manages a submenu.
+///  * [MenuItemGroup], a menu item which collects its members into a group
+///    separated from other menu items by a divider.
+///  * [MenuBarItem], a leaf menu item which displays the label, an optional
+///    shortcut label, and optional leading and trailing icons.
+///  * [MenuBarController], a class that allows closing of menus from outside of
+///    the menu bar.
+///  * [PlatformMenuBar], which creates a menu bar that is rendered by the host
+///    platform instead of by Flutter (on macOS, for example).
+///  * [ShortcutRegistry], a registry of shortcuts that apply for the entire
+///    application, used by the `MenuBar` to register its shortcuts.
+class MenuBar extends PlatformMenuBar {
+  /// Creates a const [MenuBar].
+  const MenuBar({
+    super.key,
+    this.controller,
+    this.enabled = true,
+    this.backgroundColor,
+    this.height,
+    this.padding,
+    this.elevation,
+    super.menus = const <MenuItem>[],
+  }) : _isPlatformMenu = false;
+
+  // A private constructor for the MenuBar.adaptive factory constructor to use.
+  const MenuBar._({
+    super.key,
+    this.controller,
+    this.enabled = true,
+    this.backgroundColor,
+    this.height,
+    this.padding,
+    this.elevation,
+    bool isPlatformMenu = false,
+    super.menus = const <MenuItem>[],
+  }) : _isPlatformMenu = isPlatformMenu;
+
+  /// Creates an adaptive [MenuBar] that renders using platform APIs with a
+  /// [PlatformMenuBar] on platforms that support it, currently only macOS, and
+  /// using Flutter rendering on platforms that don't (everywhere else).
+  ///
+  /// Some aspects of [MenuBar] are ignored when they are rendered by the
+  /// platform, such as any visual attributes (geometry, colors). Platform menus
+  /// also can't be closed with [MenuBarController.closeAll].
```

Comment:
Do the other features of the MenuBarController work? i.e. checking whether a menu is open?

Reply:
Checking to see if a menu is open should work. Closing the menus might be dependent upon the platform.